src/rabbit_delayed_message.erl

```diff
@@ -185,18 +187,7 @@ internal_delay_message(CurrTimer, Exchange, Delivery, Delay) ->
                        make_delay(DelayTS, Exchange, Delivery)),
     case erlang:read_timer(CurrTimer) of
         false ->
-            %% last timer expired, we set a new timer for
-            %% the next message to be delivered
-            case mnesia:dirty_first(?INDEX_TABLE_NAME) of
-                %% destructuring to prevent matching '$end_of_table'
-                #delay_key{timestamp = FirstTS} = Key
-                  when FirstTS < DelayTS ->
-                    %% there are messages that expired and need to be delivered
-                    {ok, start_timer(FirstTS - Now, Key)};
-                _ ->
-                    %% empty table or DelayTS <= FirstTS
-                    {ok, start_timer(Delay, make_key(DelayTS, Exchange))}
-            end;
+            {ok, CurrTimer};
```

Comment:
I'm probably missing something, but I don't see how that would work.

As I understand it, there are two cases where we hit this code path.

The first case is where no timer has ever been started and we need to start one to get the system moving.  I thought that my changes to the `go` handler would address that.  I don't fully understand how the plugin lifecycle works in all scenarios, so I'm willing to take your word for it that my change does not fix it.

The second case is where there was a timer, but it has expired because the message(s) at the head of the queue are ready for publication.  The timer will be restarted when the timeout handler is called, but in the meantime there could be many requests to publish messages.  I want to avoid starting a timer in that case, since it makes the publication code path very slow.  Also, those extra timers probably aren't good for performance, either.

My PR was intended to fix the second case while not breaking the first.  Based on your comment, it sounds like I have actually broken the first.  Your suggestion might help fix the first case, but I don't see how it applies to the second.

----

Perhaps I need to take the more complicated approach I had considered earlier.  We could store an extra boolean in the state that tells us whether or not a timer has been started.  That would help differentiate the two cases so we could handle them differently, though it is a little more complicated.

It's not my favorite solution, but I think it could work.  Thoughts?